backend/app/controllers/merge_request.rb

```diff
@@ -227,52 +279,137 @@ def merge_details(target, victim, selections, dry_run)
         end
         path_fix.push(part)
       end
-      path_fix_length = path_fix.length
-      if path_fix[0] != 'related_agents' && path_fix[0] != 'external_documents' && path_fix[0] != 'notes'
-        case path_fix_length
-          when 1
-            target[path_fix[0]] = victim[path_fix[0]]
-          when 2
-            target[path_fix[0]][path_fix[1]] = victim[path_fix[0]][path_fix[1]]
-          when 3
-            begin
-              if target[path_fix[0]].length <= path_fix[1]
-                target[path_fix[0]].push(victim[path_fix[0]][path_fix[1]])
-              end
-              target[path_fix[0]][path_fix[1]][path_fix[2]] = victim[path_fix[0]][path_fix[1]][path_fix[2]]
-            rescue
-              if target[path_fix[0]] === []
-                target[path_fix[0]].push(victim[path_fix[0]][path_fix[1]])
-              end
-            end
-          when 4
-            target[path_fix[0]][path_fix[1]][path_fix[2]][path_fix[3]] = victim[path_fix[0]][path_fix[1]][path_fix[2]][path_fix[3]]
-          when 5
-            begin
-              target[path_fix[0]][path_fix[1]][path_fix[2]][path_fix[3]][path_fix[4]] = victim[path_fix[0]][path_fix[1]][path_fix[2]][path_fix[3]][path_fix[4]]
-            rescue
-              if target[path_fix[0]] === []
-                target[path_fix[0]].push(victim[path_fix[0]][path_fix[1]])
-              elsif target[path_fix[0]][path_fix[1]][path_fix[2]] === []
-                target[path_fix[0]][path_fix[1]][path_fix[2]].push(victim[path_fix[0]][path_fix[1]][path_fix[2]][path_fix[3]])
-              end
-            end
-        end
-      elsif path_fix[0] === 'external_documents'
-        target['external_documents'].push(victim['external_documents'][path_fix[1]])
-      elsif path_fix[0] === 'notes'
-        target['notes'].push(victim['notes'][path_fix[1]])
+
+      subrec_name = path_fix[0]
+      victim_values[subrec_name] = values_from_params[subrec_name]
+
+      # subrec level add/replace 
+      if path_fix[2] == "append" || path_fix[2] == "replace"
+        subrec_add_replacements.push(path_fix)
+
+      # field level replace
+      else
+        field_replacements.push(path_fix)
       end
-      target['title'] = target['names'][0]['sort_name']
-    end
-    if dry_run == true
```

Comment:
@avatar382 I am not seeing where dry_run is used. I believe that used to be preview. Is that handled somewhere else now? If so, is dry_run needed in the call?